lib/rspec/matchers/built_in/yield.rb

```diff
@@ -182,10 +194,37 @@ def supports_block_expectations?
       private
 
         def set_expected_yields_count(relativity, n)
-          raise "Multiple count constraints are not supported" if @expectation_type
+          raise_unsupported_yield_expectation if unsupported_yield_expectation?(relativity)
+
+          constraint_count = count_constraint_to_number(n)
 
-          @expectation_type = relativity
-          @expected_yields_count = count_constraint_to_number(n)
+          if @expectation_type == :<= && relativity == :>=
```

Comment:
What do you think about adding a check so that `at_least(3).at_most(2)` can't be called?